examples/azure/chat_with_your_own_data.ipynb

```diff
@@ -0,0 +1,432 @@
+    "        req.headers[\"Authorization\"] = f\"Bearer {self.cached_token.token}\"\n",
+    "        return req\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Chat completion model with your own data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To chat with Azure OpenAI models using your own data with the Python SDK, we must first set up the code to target the chat completions extensions endpoint which is designed to work with your own data. To do this, we've created a convenience function that can be called to set a custom adapter for the library which will target the extensions endpoint for a given deployment ID."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "\n",
+    "def setup_byod(deployment_id: str) -> None:\n",
+    "    \"\"\"Sets up the OpenAI Python SDK to use your own data for the chat endpoint.\n",
+    "    \n",
+    "    :param deployment_id: The deployment ID for the model to use with your own data.\n",
+    "\n",
+    "    To remove this configuration, simply set openai.requestssession to None.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    class BringYourOwnDataAdapter(requests.adapters.HTTPAdapter):\n",
+    "\n",
+    "        def send(self, request, **kwargs):\n",
+    "            request.url = f\"{openai.api_base}/openai/deployments/{deployment_id}/extensions/chat/completions?api-version={openai.api_version}\"\n",
+    "            return super().send(request, **kwargs)\n",
+    "\n",
+    "    session = requests.Session()\n",
+    "\n",
+    "    # Mount a custom adapter which will use the extensions endpoint for any call using the given `deployment_id`\n",
+    "    session.mount(\n",
+    "        prefix=f\"{openai.api_base}/openai/deployments/{deployment_id}\",\n",
+    "        adapter=BringYourOwnDataAdapter()\n",
+    "    )\n",
+    "\n",
+    "    if use_azure_active_directory:\n",
+    "        session.auth = TokenRefresh(default_credential, [\"https://cognitiveservices.azure.com/.default\"])\n",
+    "\n",
+    "    openai.requestssession = session\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can call the convenience function to configure the SDK with the model we plan to use for our own data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "setup_byod(\"gpt-4\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Providing our search endpoint, key, and index name for the `dataSources` keyword argument, any questions posed to the model will now be grounded in your own data. An additional property, `context`, will be provided to show the data the model referenced to answer the question."
```

Comment:
to me it is not very clear what data this example is using. it would be great if we could provide an easy to use example endpoint, or more information about what is actually going under the hood here